project/driver_controls/src/driver_io.c

```diff
@@ -0,0 +1,113 @@
+#include <stddef.h>
+
+#include "driver_io.h"
+#include "gpio_it.h"
+#include "input_interrupt.h"
+
+static const DriverIO driver_io[NUM_DRIVER_IO_INPUTS] = {
```

Comment:
What's the point of this array? (Also, you have some scope issues - this should be prefixed with `s_`.)